query-engine/core/src/query_graph_builder/write/create.rs

```diff
@@ -107,7 +114,93 @@ pub fn create_record_node(
 
     args.add_datetimes(&model);
 
-    let cr = CreateRecord { model, args };
+    let selected_fields = model.primary_identifier();
+    let selection_order = selected_fields.db_names().collect();
+
+    let cr = CreateRecord {
+        // A regular create record is never used as a result node. Therefore, it's never serialized, so we don't need a name.
+        name: String::new(),
```

Comment:
Yup, this is immutable and creating an empty string has zero cost. I was wondering if given the name is not always necessary and should be an Option<String> but it's perfectly fine and more straightforward to be this way.

Reply:
The issue with having an `Option<String>` for the name is that eventually, that CreateRecord will be transformed into a [RecordSelection](https://github.com/prisma/prisma-engines/blob/a651a3274a6dd554823daff4f94fccb1c99b7ac9/query-engine/core/src/result_ast/mod.rs#L16), which requires a name (even though the ones that don't contain names won't end up serialized).

I thought about creating an enum of `CreateRecord { Atomic(), Regular }` to make that explicit and either return a `QueryResult::Id` or a `QueryResult::RecordSelection` based on that, but I didn't want to make the interface more complex with different types of creates to handle.